docs/vocabulary.md

```diff
@@ -0,0 +1,312 @@
+
+- **Step**: Basic building block to execute one or more commands.
+- **Stage**: Collection of steps and the required configuration.
+- **Pipeline**: Collection of one or more stages to run.
+- **Agent**: Container image or Jenkins agent to schedule pipelines on.
+- **Trigger**: A pipeline trigger defines when to automatically run a pipeline.
+
+### Screwdriver
+
+[Screwdriver](https://screwdriver.cd) is an open source build platform designed
+for Continuous Delivery.
+
+Some of the core Screwdriver terms are listed below. [[6]]
+
+- **Step**: List of commands to execute.
+- **Job**: Collection of the steps to run.
+- **Workflow**: Collection and order of jobs to run.
+- **Pipeline**: Used interchangeably with workflow. (*citation needed*)
+- **Trigger**: A pipeline trigger defines when to automatically run a pipeline
+or job. (*citation needed*)
+
+### Spinnaker
+
+[Spinnaker](https://www.spinnaker.io/) is an open source, multi-cloud continuous
+delivery platform for releasing software changes with high velocity and
+confidence.
+
+Some of the core Spinnaker terms are listed below. [[7]]
+
+- **Task**: An atomic function to perform.
+- **Stage**: A collection of sequential Tasks and composed Stages that describe
+a higher-level action the Pipeline will perform either linearly or in parallel.
+- **Pipeline**: The pipeline is the key deployment management construct in
+Spinnaker. It consists of a sequence of actions, known as stages.
+- **Pipeline Template**: A parameterized pipeline, minus the pipeline
+configuration found on a pipeline instance. This template helps developers
+create pipelines that follow a pattern that you establish.
+- **Pipeline configuration**: The same as the configuration for a pipeline
+not created from a template, plus variable bindings and a reference to the
+template.
+- **Cluster**: Logical groupings of Server Groups in Spinnaker.
+- **Provider**: A Cloud Provider is an interface to a set of virtual resources
+that Spinnaker has control over.
+- **Trigger**: A pipeline trigger defines when to automatically run a pipeline.
+
+### Tekton
+
+[The Tekton Pipelines](https://github.com/tektoncd/pipeline) project provides
+k8s-style resources for declaring CI/CD-style pipelines.
+
+Some of the core Tekton terms are listed below. [[8]]
+
+- **Step**: a specific function to perform.
+- **Task**: is a collection of sequential steps you would want to run as part of your
+continuous integration flow. A task will run inside a pod on your cluster.
+- **ClusterTask**: Similar to Task, but with a cluster scope.
+- **Pipeline**: stateless, reusable, paramterized collection of tasks. Tasks linked
+- **PipelineRun**: running instantiations of Pipelines
+by input and output PipelineResources
+- **Pipeline Resource**: input to or output from tasks
+- **Trigger**: Triggers is a Kubernetes Custom Resource Defintion (CRD) controller
+that allows you to extract information from events payloads (a "trigger") to create
+Kubernetes resources.
```

Comment:
At the moment there is actually no "Trigger" concept, what we do have is:

1. TriggerTemplates - Templates that describe how to instantiate resources such as PipelineRuns with parameters
2. TriggerBindings - These describe how to extract values from a json structure using JSON path and use them to provide parameters (e.g. to TriggerTemplates)
3. EventListeners - Correspond to running endpoints that event systems can call with json payloads

Reply:
ditto